### PR TITLE
Add error and success Atom notifications

### PR DESCRIPTION
**Pull request by xixixao**
This is the last major thing I wanted to do here as outlined in https://github.com/facebooknuclide/nuclide-format-js/pull/17. If the file includes a syntax error, instead of silently failing, we show a notification. If the transform produces no changes, we show a short-lived success notification.

Most of the code here makes sure we only show one notification at a time.

I could very easily write a pretty printer for the error message, but I'd rather have it inline (I think we do it now for prettier).

screen shot 2017-06-19 at 01 31 45
screen shot 2017-06-19 at 01 31 37

### CODE REVIEW

File: src/atom/formatCode.js [hunk 1]
```diff
@@ -49,4 +47,63 @@ async function formatCode(options: SourceOptions, editor_: ?TextEditor): Promise
...
+    return inputSource;
+  }
+  dismissExistingErrorNotification();
+  if (outputSource === inputSource) {
+    showSuccessNotification();
+  }
+  return outputSource;
+}
+
+const ERROR_TITLE = 'nuclide-format-js failed';
```
Comment (kyldvs) [inline src/atom/formatCode.js]: nuclide-format-js is a bad name, I wonder if here we could say something like:

```
(nuclide-format-js) Add requires failed
```

File: src/atom/formatCode.js [hunk 2]
```diff
@@ -49,4 +47,63 @@ async function formatCode(options: SourceOptions, editor_: ?TextEditor): Promise
...
+    stack: error.stack,
+    dismissable: true,
+  });
+}
+
+function dismissExistingErrorNotification(): void {
+  dismissNotification(ERROR_TITLE);
+}
+
+const SUCCESS_TITLE = 'nuclide-format-js succeeded';
```
Comment (kyldvs) [inline src/atom/formatCode.js]: And something similar here as well

**Comment by kyldvs** [comment]
Does it make sense to show this on success? Maybe make that an option? I could see it being annoying for those of us who hit these commands frequently while coding

**Comment by xixixao** [comment]
@kyldvs I added an opt out. The success notification only shows if there were no changes. This is important because otherwise the user gets no feedback: The command succeed, or did it just silently fail?

screen shot 2017-06-19 at 23 20 58
screen shot 2017-06-19 at 23 21 10

**Comment by xixixao** [comment]
So I realized that prettier doesn't show syntax errors inline. But showing the code in the notification sucks, so I move the cursor to the location instead. @vjeux you might want to do this in prettier too (along with the removal of previous error notifications) - at least until I can go back to the lenient parser project :)

screen shot 2017-06-19 at 23 45 16

**Pull request merged**